lock/lock.go

```diff
@@ -108,19 +108,26 @@ func (d *Driver) ExecuteWithMultiUseLock(ucs []config.UseLocker, timeout time.Du
 // lock after a timeout, returning a stop channel. Timeout is jittered to
 // mitigate thundering herd problems.
 func (d *Driver) AcquireWithTTLRefresh(uc config.UseLocker, ttl, timeout time.Duration) (chan struct{}, error) {
+	if err := d.Config.PublishUse(uc); err != nil {
+		return nil, err
+	}
+
 	if err := d.acquire(uc, ttl, timeout); err != nil {
```

Comment:
please provide an example (either code or description) which demonstrates the issues you are predicting/seeing.